### PR TITLE
New generate reset key subcommand and minor fixes

### PR DESCRIPTION
**Pull request by nick-child-ibm**
This PR is to update minor bugs and minor enhancements. These include:
  -Fixing a typo of the default path to secure variables 
  -Edited compiler optimization from -Os to -O2 because -Os was causing issues on ppc (opened a gcc bug report [here](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98692))
  -Allow for custom man page destination during install with `make install MANDIR=<path>`

I also created an easy way to generate key reset files with `secvarctl generate reset -n <varName> -k <signerKey> -c <signerCrt>`
This update has been reflected in documentation.

### CODE REVIEW

File: secvarctl.c [hunk 1]
```diff
@@ -123,7 +123,7 @@ static struct backend *getBackend()
 		prlog(PR_WARNING, "WARNING!! Could not extract data from %s , assuming platform does not support secure variables\n", secVarFormatLocation);
 		goto out;
 	}
-	//loop through all known backends
+	// loop through all known backends
```
Comment (richteer) [inline secvarctl.c:126]: This commit does not need to be in the history. If you want to keep this comment change, change the description instead to reflect what it is doing.

File: README.md [hunk 2]
```diff
@@ -191,6 +196,8 @@ For any questions regarding secvarctl, feel free to reach out: [Nick Child](nick
 		A PKCS7 and Auth file can be signed with several signers by adding more ' -k <privKey> -c <cert>' pairs. 
 		Additionaly, when generating an Auth file the secure variable name must be given as -n <keyName> because it is included in the  message digest. 
 		When using the input type '[f]ile' it will be assumed to be a text file and if output file is '[e]sl', '[p]kcs7' or '[a]uth' it will be hashed according to <hashAlg> (default SHA256). 
+		To create a variable reset file (one that will remove the current contents of a variable), replace '<inputFormat>:<outputFormat>' with 'reset' and
+		supply a variable name, public and private signer files and an output file with '-n <keyName> -k <privKey> -c <crtFile> -o <outFile>'
 		GENERATION OF PKCS7 AND AUTH FILES ARE IN EXPERIMENTAL DEVELEPOMENT PHASE. THEY HAVE NOT BEEN THOROUGHLY TESTED YET.
```
Comment (richteer) [inline README.md:203]: Commit message should include a little more about what is being updated about this file. It seems like it is adding documentation for the `reset` command? Not sure what changed about the blob at the top, probably whitespace?

File: Makefile [hunk 3]
```diff
@@ -46,8 +46,6 @@ else
 	CRYPTO_OBJ = $(patsubst %,$(SKIBOOTOBJDIR)/%, $(_CRYPTO_OBJ))
 	OBJ += $(CRYPTO_OBJ)
 endif
-# OBJDIR=obj
-# OBJ=$(patsubst %, $(OBJDIR)/%,$(_OBJ))
```
Comment (richteer) [inline Makefile:49]: Again, please don't include Jenkins-related commits in the history

File: backends/edk2-compat/include/edk2-svc.h [hunk 4]
```diff
@@ -13,7 +13,7 @@
 #define CERT_BUFFER_SIZE        2048
 
 #ifndef SECVARPATH
-#define SECVARPATH "/sys/firmware/secvar/vars"
+#define SECVARPATH "/sys/firmware/secvar/vars/"
 #endif
```
Comment (richteer) [inline backends/edk2-compat/include/edk2-svc.h:18]: This short commit message is fine, since it's literally a one character change.

File: Makefile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #_*_MakeFile_*_
 CC = gcc 
-_CFLAGS = -Os -s -g -std=gnu99
+_CFLAGS = -O2 -s -g -std=gnu99
```
Comment (richteer) [inline Makefile:3]: This should be its own commit, with its own description explaining why this change was made. If this needed to be bisected by someone else, knowing why this change was made may make a big difference, which is why it shouldn't be lumped in with another change.

File: secvarctl.c (hunk 1 above)
Comment (erichte-ibm) [inline secvarctl.c:126]: Don't include Jenkins-related commits in the history here. If you want to keep this change, have the commit message reflect what is actually changing in the commit.

File: README.md (hunk 2 above)
Comment (erichte-ibm) [inline README.md:203]: General commit message comment: Specify what updated about the readme. Looks like the reset command was added? Something else changed about the whitespace i guess in that other list?

File: Makefile [hunk 6]
```diff
@@ -46,8 +46,6 @@ else
 	CRYPTO_OBJ = $(patsubst %,$(SKIBOOTOBJDIR)/%, $(_CRYPTO_OBJ))
 	OBJ += $(CRYPTO_OBJ)
 endif
-# OBJDIR=obj
-# OBJ=$(patsubst %, $(OBJDIR)/%,$(_OBJ))
```
Comment (erichte-ibm) [inline Makefile:50]: Again, please try not to include Jenkins commits in the final commit history.

File: backends/edk2-compat/include/edk2-svc.h [hunk 7]
```diff
@@ -13,7 +13,7 @@
 #define CERT_BUFFER_SIZE        2048
 
 #ifndef SECVARPATH
-#define SECVARPATH "/sys/firmware/secvar/vars"
+#define SECVARPATH "/sys/firmware/secvar/vars/"
```
Comment (erichte-ibm) [inline backends/edk2-compat/include/edk2-svc.h:16]: Note: this short commit message is fine, since it's literally a 1 character typo fix that isn't largely consequential.

File: Makefile (hunk 5 above)
Comment (erichte-ibm) [inline Makefile:3]: This one-character change however, is very consequential. This should be its own commit, with a longer detailed message explaining why this change is mad. Consider someone else trying to debug an issue, and running bisect -- this may be important information to someone in the future using an old GCC wondering why they are seeing the issue again, etc.

**Review by erichte-ibm — CHANGES_REQUESTED** [review]
Aside from the Jenkins commits, the rest largely looks fine.

The major things I would like changed before merge:
 - Remove the Jenkins commits (or merge them and call it a "formatting fix" instead)
 - Split out the optimization change into its own commit

In the future, I'd prefer some of the larger commits separated out -- try not to merge fixes with features. Fixes should be their own commits, even if they are small, as they provide documentation in the git history on when a bug was addressed.
